### PR TITLE
fix(expr): evaluate scalar expr non-strictly in streaming

### LINKED ISSUE #12474: bug: scalar expressions in `StreamProjectSet` are not evaluated non-strictly
### Describe the bug

Scalar expressions in `StreamProjectSet` are not evaluated non-strictly, where failure will lead to recovery. This is inconsistent with the behavior of `Project`.

### Error message/log

failed to execute barrier: Rpc error: gRPC error (Internal error): Actor 72 exit unexpectedly: Executor error: Chunk operation error: Division by zero

### To Reproduce

```sql
create table t(x int[]);
insert into t values (array[1,2]);

create materialized view mv as select x[1]/0 as bomb, unnest(x) as unnest from t;
```

### Expected behavior

```sql
dev=> select * from mv2;
 bomb | unnest 
------+--------
      |      1
      |      2
```

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

fix #12474

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/project_set.rs [hunk 1]
```diff
@@ -253,3 +252,61 @@ impl Inner {
         Ok(ret)
     }
 }
+
+/// Either a scalar expression or a set-returning function.
+///
+/// See also [`PbProjectSetSelectItem`]
```
Comment (BugenZhao) [inline src/stream/src/executor/project_set.rs:258]: Can we also slightly mention how this differs to the one defined in `batch`?

**Review by BugenZhao — APPROVED** [review]
LGTM

**Comment by wangrunji0408** [comment]
@BugenZhao any ideas on how to rewrite `e2e_test/error_ui/simple/recovery.slt`?

**Comment by BugenZhao** [comment]
> @BugenZhao any ideas on how to rewrite `e2e_test/error_ui/simple/recovery.slt`? one option is to switch to another bug #11915 for now. 🤪

Sure. Let's use that as a workaround until we find a good approach to reliably trigger a meaningful failure.

**Comment by wangrunji0408** [comment]
What about introducing a test-only function `rw_panic()`?

**Comment by BugenZhao** [comment]
> What about introducing a test-only function `rw_panic()`?

We already have a `RECOVERY` command to tigger a recovery on _meta_, and I've also attempted to add a command to trigger a failure on _compute node_ (https://github.com/risingwavelabs/risingwave/pull/16397#discussion_r1571899490).

However, as this test also covers the root cause identification (https://github.com/risingwavelabs/risingwave/pull/16431), I'm afraid whether a manually-triggered failure is general enough. 🤔

**Pull request merged**